sw/host/opentitanlib/src/test_utils/lc_transition.rs

```diff
@@ -0,0 +1,146 @@
+    let lc_status = jtag.read_lc_ctrl_reg(&LcCtrlReg::Status)?;
+    let expected_lc_status =
+        LcCtrlStatusBit::union([LcCtrlStatusBit::Initialized, LcCtrlStatusBit::Ready]);
+    if lc_status != expected_lc_status {
+        return Err(LcTransitionError::LcCtrlNotReady(lc_status).into());
+    }
+
+    // Check we are under the maximum LC transition count thresholds.
+    let lc_transition_cnt = jtag.read_lc_ctrl_reg(&LcCtrlReg::LcTransitionCnt)?;
+    if lc_transition_cnt >= MAX_LC_TRANSITIONS {
+        return Err(LcTransitionError::MaxTransitionCountExceeded(lc_transition_cnt).into());
+    }
+
+    // Check the LC transition mutex has not been claimed yet.
+    if jtag.read_lc_ctrl_reg(&LcCtrlReg::ClaimTransitionIf)? == u8::from(MultiBitBool8::True) as u32
+    {
+        return Err(LcTransitionError::MutexAlreadyClaimed.into());
+    }
+
+    // Attempt to claim the LC transition mutex.
+    jtag.write_lc_ctrl_reg(
+        &LcCtrlReg::ClaimTransitionIf,
+        u8::from(MultiBitBool8::True) as u32,
+    )?;
+
+    // Check the LC transition mutex was claimed.
+    if jtag.read_lc_ctrl_reg(&LcCtrlReg::ClaimTransitionIf)? != u8::from(MultiBitBool8::True) as u32
+    {
+        return Err(LcTransitionError::FailedToClaimMutex.into());
+    }
+
+    // Program the target LC state, i.e., Scrap.
+    jtag.write_lc_ctrl_reg(
+        &LcCtrlReg::TransitionTarget,
+        target_lc_state.redundant_encoding(),
+    )?;
+
+    // Check correct target LC state was programmed.
+    let target_lc_state_programmed = jtag.read_lc_ctrl_reg(&LcCtrlReg::TransitionTarget)?;
+    if target_lc_state_programmed != target_lc_state.redundant_encoding() {
+        return Err(LcTransitionError::TargetProgrammingFailed(target_lc_state_programmed).into());
+    }
+
+    // Configure external clock.
+    if use_external_clk {
+        jtag.write_lc_ctrl_reg(
+            &LcCtrlReg::TransitionCtrl,
+            LcCtrlTransitionCtrlBit::union([LcCtrlTransitionCtrlBit::ExtClockEn]),
+        )?;
+    } else {
+        jtag.write_lc_ctrl_reg(&LcCtrlReg::TransitionCtrl, 0)?;
+    }
+
+    // Check external clock was not requested.
+    assert_ne!(
+        jtag.read_lc_ctrl_reg(&LcCtrlReg::TransitionCtrl)?,
+        LcCtrlTransitionCtrlBit::union([LcCtrlTransitionCtrlBit::ExtClockEn]),
+    );
+
+    // Initiate LC transition and poll status register until transition is completed.
+    jtag.write_lc_ctrl_reg(
+        &LcCtrlReg::TransitionCmd,
+        LcCtrlTransitionCmdBit::union([LcCtrlTransitionCmdBit::Start]),
+    )?;
+    loop {
+        let status = jtag.read_lc_ctrl_reg(&LcCtrlReg::Status)?;
+        if (status & LcCtrlStatusBit::union([LcCtrlStatusBit::TransitionSuccessful])) != 0 {
+            break;
+        }
+        if (status
+            & !LcCtrlStatusBit::union([
+                LcCtrlStatusBit::Initialized,
+                LcCtrlStatusBit::Ready,
+                LcCtrlStatusBit::TransitionSuccessful,
+            ]))
+            != 0
```

Comment:
Would be nice to use [bitflags](https://docs.rs/bitflags/latest/bitflags/) for these status bits to avoid bit checking tricks like this, but beyond the scope of this PR.

Reply:
bitflags is a popular dependency for doing these bit-mask checks, so something like this would replace our `LcBit` trait:

```rust
use bitflags::bitflags;

bitflags! {
    #[derive(Clone, Copy, Debug, PartialEq, Eq)]
    struct LcCtrlStatus: u32 {
        const INITIALIZED            = 0b1 << bindgen::dif::LC_CTRL_STATUS_INITIALIZED_BIT;
        const READY                  = 0b1 << bindgen::dif::LC_CTRL_STATUS_READY_BIT;
        const TRANSITION_SUCCESSFUL  = 0b1 << bindgen::dif::LC_CTRL_STATUS_TRANSITION_SUCCESSFUL_BIT;
        const TRANSITION_COUNT_ERROR = 0b1 << bindgen::dif::LC_CTRL_STATUS_TRANSITION_COUNT_ERROR_BIT;
        const TRANSITION_ERROR       = 0b1 << bindgen::dif::LC_CTRL_STATUS_TRANSITION_ERROR_BIT;
        const TOKEN_ERROR            = 0b1 << bindgen::dif::LC_CTRL_STATUS_TOKEN_ERROR_BIT;
        const FLASH_RMA_ERROR        = 0b1 << bindgen::dif::LC_CTRL_STATUS_FLASH_RMA_ERROR_BIT;
        const OTP_ERROR              = 0b1 << bindgen::dif::LC_CTRL_STATUS_OTP_ERROR_BIT;
        const STATE_ERROR            = 0b1 << bindgen::dif::LC_CTRL_STATUS_STATE_ERROR_BIT;
        const BUS_INTEG_ERROR        = 0b1 << bindgen::dif::LC_CTRL_STATUS_BUS_INTEG_ERROR_BIT;
        const OTP_PARTITION_ERROR    = 0b1 << bindgen::dif::LC_CTRL_STATUS_OTP_PARTITION_ERROR_BIT;
    }
}

let expected = LcCtrlStatus::INITIALIZED | LcCtrlStatus::READY | LcCtrlStatus::TRANSITION_SUCCESSFUL;
let status = LcCtrlStatus::from_bits(status).expect("invalid bits set");

if status.contains(expected) {
    // ...
}
```

Again, this is just something I'd like to see opentitanlib use to avoid confusing bitflag checks like this one, but out of scope for this PR.

Reply:
Ah yeah that would be a bit more ergonomic, but would require a refactor of the current code in dif/lc_ctrl.rs. Perhaps in the future.